src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityServiceTest.kt

```diff
@@ -0,0 +1,222 @@
+import java.time.Clock
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+
+class EligibilityServiceTest {
+
+  private val service =
+    EligibilityService(
+      clock,
+    )
+
+  @Test
+  fun `Person is eligible for CVL`() {
+    val result = service.isEligibleForCvl(aPrisonerSearchResult)
+    assertThat(result).isTrue()
+  }
+
+  @Test
+  fun `Person is parole eligible but parole eligibility date is in the past - eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        paroleEligibilityDate = LocalDate.now(clock).minusDays(1),
+      ),
+    )
+    assertThat(result).isTrue()
+  }
+
+  @Test
+  fun `Person is parole eligible - not eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        paroleEligibilityDate = LocalDate.now(clock).plusYears(1),
+      ),
+    )
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun `Person has an incorrect legal status - not eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        legalStatus = "DEAD",
+      ),
+    )
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun `Person is on an indeterminate sentence - not eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        indeterminateSentence = true,
+      ),
+    )
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun `Person does not have a conditional release date - not eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        conditionalReleaseDate = null,
+      ),
+    )
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun `Person is on ineligible EDS - ARD is outside threshold in the past - not eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        confirmedReleaseDate = LocalDate.now(clock).minusDays(5),
+      ),
+    )
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun `Person is on ineligible EDS - ARD is outside threshold in the future - not eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        paroleEligibilityDate = LocalDate.now(clock).minusDays(1),
+        confirmedReleaseDate = LocalDate.now(clock).plusDays(2),
+      ),
+    )
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun `Person is on ineligible EDS - has a APD and a PED in the past - not eligible for CVL `() {
+    val result = service.isEligibleForCvl(
+      aPrisonerSearchResult.copy(
+        paroleEligibilityDate = LocalDate.now(clock).minusDays(1),
+        actualParoleDate = LocalDate.now(clock).plusDays(1),
+      ),
+    )
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun `Person is on ineligible EDS - has a APD with a PED in the past - not eligible for CVL `() {
```

Comment:
I think this test may be a duplicate of the one above

Reply:
Changed